workbench-core/authentication/src/tokenRevocationService.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DynamoDBService, buildDynamoDBPkSk } from '@aws/workbench-core-base';
+import jwtDecode from 'jwt-decode';
+import _ from 'lodash';
+import { InvalidJWTError } from './errors/invalidJwtError';
+
+interface TokenRevocationRecord {
+  key: {
+    pk: string;
+    sk: string;
+  };
+  item: {
+    ttl: string;
+  };
+}
+/**
+ * Service used to revoke tokens
+ */
+export class TokenRevocationService {
+  private readonly _tokenRevocationType: string = 'REVOKEDTOKEN';
+  private _ddbService: DynamoDBService;
+  public constructor(options: {
+    dynamoDBSettings: {
+      region: string;
+      table: string;
+    };
+  }) {
+    const { dynamoDBSettings } = options;
+    this._ddbService = new DynamoDBService(dynamoDBSettings);
+  }
+  /**
+   * Revoke given token
+   * @param revokeTokenRequest - requires the token to be revoked
+   */
+  public async revokeToken(revokeTokenRequest: { token: string }): Promise<void> {
+    const { token } = revokeTokenRequest;
+    const { key, item } = this._transformToTokenRevocationRecord(token);
+
+    await this._ddbService.updateExecuteAndFormat({
+      key,
+      params: {
+        item
+      }
+    });
+  }
+  /**
+   * Checks if a given token is revoked
+   * @param isRevokedRequest - requires the token to be checked
+   * @returns
+   */
+  public async isRevoked(isRevokedRequest: { token: string }): Promise<boolean> {
+    const { token } = isRevokedRequest;
+
+    const { key } = this._transformToTokenRevocationRecord(token);
+    const item = await this._ddbService.getItem({
+      key
+    });
+    return !!item;
```

Comment:
Maybe I'm missing something, but from line 74, an item is 
```
{
  ttl: <expiration time as a number>
}
```

If that's the case would line 62 return a boolean? Don't we need to compare the `ttl` with the current time?

Reply:
The ttl is meant for DDB to remove the entry, once token is revoked, it should stay revoked so there is no need to compare the TTL